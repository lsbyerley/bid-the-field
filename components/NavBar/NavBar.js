import Link from 'next/link';
import { useSession, signIn, signOut } from 'next-auth/react';
import { MenuAlt1Icon } from '@heroicons/react/outline';
import ThemeSwitch from '../ThemeSwitch';

const NavBar = () => {
  const { data: session, status: sessionStatus } = useSession();
  const sessionLoading = sessionStatus === 'loading';

  console.log('LOG: session', session);

  return (
    <div className='navbar bg-base-200'>
      <div className='navbar-start'>
        <div className='dropdown'>
          <label tabIndex='0' className='btn btn-ghost lg:hidden'>
            <MenuAlt1Icon className='w-5 h-5' />
          </label>
          <ul
            tabIndex='0'
            className='p-2 mt-3 shadow menu menu-compact dropdown-content bg-base-200 rounded-box w-52'
          >
            <li>
              <a>About</a>
            </li>
          </ul>
        </div>
        <Link href='/'>
          <a className='text-xl normal-case btn btn-ghost'>Bid The Field</a>
        </Link>
      </div>
      <div className='hidden navbar-center lg:flex'>
        <ul className='p-0 menu menu-horizontal'>
          <li>
            <a>About</a>
          </li>
        </ul>
      </div>
      <div className='navbar-end'>
        <ThemeSwitch />
        {!sessionLoading && !session && (
          <div>
            <a
              href={`/api/auth/signin`}
              className='ml-4 btn btn-sm'
              onClick={(e) => {
                e.preventDefault();
                signIn();
              }}
            >
              Sign in
            </a>
          </div>
        )}
        {!sessionLoading && session && (
          <div className='dropdown dropdown-end'>
            <label tabIndex='0' className='btn btn-ghost btn-circle avatar'>
              <div className='w-10 rounded-full'>
                <img
                  referrerPolicy='no-referrer'
                  src={session?.user?.image || 'https://place-hold.it/40x40'}
                />
              </div>
            </label>
            <ul
              tabIndex='0'
              className='p-2 mt-3 shadow menu menu-compact dropdown-content bg-base-200 rounded-box w-52'
            >
              <li className='disabled'>
                <span className=''>{session.user.email}</span>
              </li>
              <li>
                <a
                  href={`/api/auth/signout`}
                  onClick={(e) => {
                    e.preventDefault();
                    signOut();
                  }}
                >
                  Sign out
                </a>
              </li>
            </ul>
          </div>
        )}
      </div>
    </div>
  );
};

export default NavBar;
