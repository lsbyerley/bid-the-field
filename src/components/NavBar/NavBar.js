import Link from 'next/link';
import { useUser } from '@supabase/auth-helpers-react';
import { supabaseClient } from '@supabase/auth-helpers-nextjs';
import { MenuAlt1Icon } from '@heroicons/react/outline';
import ThemeSwitch from '../ThemeSwitch';

const NavBar = () => {
  const { isLoading, user, error } = useUser();

  return (
    <div className='navbar bg-base-100'>
      <div className='navbar-start'>
        <div className='dropdown'>
          <label tabIndex='0' className='btn btn-sm btn-ghost lg:hidden'>
            <MenuAlt1Icon className='w-5 h-5' />
          </label>
          <ul
            tabIndex='0'
            className='p-2 mt-3 shadow menu menu-compact dropdown-content bg-base-100 rounded-box w-52'
          >
            <li>
              {/* <Link href='/about'>
                <a>About</a>
  </Link> */}
              <Link href='/'>
                <a className='active:bg-base-300'>Home</a>
              </Link>
            </li>
          </ul>
        </div>
        <Link href='/'>
          <a className='normal-case md:text-lg btn btn-sm btn-ghost'>
            Bid The Field
          </a>
        </Link>
      </div>
      <div className='hidden navbar-center lg:flex'>
        {/*<ul className='p-0 menu menu-horizontal'>
          <li>
            <Link href='/about'>
              <a className='btn btn-ghost'>About</a>
            </Link>
          </li>
</ul>*/}
      </div>
      <div className='navbar-end'>
        <ThemeSwitch />
        {!isLoading && !user && (
          <button
            className='ml-4 btn btn-sm'
            onClick={() => {
              supabaseClient.auth.signIn({ provider: 'google' });
            }}
          >
            Sign In
          </button>
        )}
        {!isLoading && user && (
          <div className='ml-2 md:ml-4 dropdown dropdown-end'>
            <label tabIndex='0' className='btn btn-ghost btn-circle avatar'>
              <div className='w-8 rounded-full md:w-10'>
                <img
                  referrerPolicy='no-referrer'
                  src={
                    user?.user_metadata?.picture ||
                    'https://place-hold.it/40x40'
                  }
                />
              </div>
            </label>
            <ul
              tabIndex='0'
              className='p-2 mt-3 shadow menu menu-compact dropdown-content bg-base-100 rounded-box w-52'
            >
              <li className='menu-title'>
                <div>{user?.email || 'no email'}</div>
              </li>
              <li>
                <Link href='/api/auth/logout'>
                  <a className='active:bg-gray-500'>Logout</a>
                </Link>
              </li>
            </ul>
          </div>
        )}
      </div>
    </div>
  );
};

export default NavBar;
