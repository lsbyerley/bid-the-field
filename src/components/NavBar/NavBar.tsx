import Link from 'next/link';
import {
  useSessionContext,
  useSupabaseClient,
} from '@supabase/auth-helpers-react';
import { useRouter } from 'next/router';
import { Bars3Icon } from '@heroicons/react/24/outline';
import ThemeSwitch from '../ThemeSwitch';

const NavBar = () => {
  const router = useRouter();
  const { isLoading, session, error } = useSessionContext();
  const supabaseClient = useSupabaseClient();

  return (
    <div className='navbar bg-base-100'>
      <div className='navbar-start'>
        <div className='dropdown'>
          <label tabIndex={0} className='btn btn-sm btn-ghost lg:hidden'>
            <Bars3Icon className='w-5 h-5' />
          </label>
          <ul
            tabIndex={0}
            className='p-2 mt-3 shadow menu menu-compact dropdown-content bg-base-100 rounded-box w-52'
          >
            <li>
              <Link href='/' className='active:bg-base-300'>
                Home
              </Link>
              <Link href='/profile' className='active:bg-base-300'>
                Profile
              </Link>
            </li>
          </ul>
        </div>
        <Link href='/' className='normal-case md:text-lg btn btn-sm btn-ghost'>
          Bid The Field
        </Link>
      </div>
      <div className='hidden navbar-center lg:flex'>
        {/*<ul className='p-0 menu menu-horizontal'>
          <li>
            <Link href='/about'>
              <a className='btn btn-ghost'>About</a>
            </Link>
          </li>
</ul>*/}
      </div>
      <div className='navbar-end'>
        <ThemeSwitch />
        {!isLoading && !session && (
          <button
            className='ml-4 btn btn-sm'
            onClick={(e) => {
              e.preventDefault();
              return supabaseClient.auth.signInWithOAuth({
                provider: 'google',
              });
            }}
          >
            Sign In
          </button>
        )}
        {!isLoading && session && (
          <div className='ml-2 md:ml-4 dropdown dropdown-end'>
            <label tabIndex={0} className='btn btn-ghost btn-circle avatar'>
              <div className='w-8 rounded-full md:w-10'>
                <img
                  referrerPolicy='no-referrer'
                  src={
                    session.user?.user_metadata?.picture ||
                    'https://place-hold.it/40x40'
                  }
                />
              </div>
            </label>
            <ul
              tabIndex={0}
              className='p-2 mt-3 shadow menu menu-compact dropdown-content bg-base-100 rounded-box w-52'
            >
              <li className='menu-title'>
                <div>
                  {session.user?.user_metadata?.name ||
                    session.user?.user_metadata?.email ||
                    'n/a'}
                </div>
              </li>
              <li>
                <Link href='/profile' className='active:bg-gray-90'>
                  Profile
                </Link>
              </li>
              <li>
                <button
                  onClick={async () => {
                    await supabaseClient.auth.signOut();
                    router.push('/');
                  }}
                >
                  Logout
                </button>
              </li>
            </ul>
          </div>
        )}
      </div>
    </div>
  );
};

export default NavBar;
