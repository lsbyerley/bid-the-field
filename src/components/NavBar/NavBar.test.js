import React from 'react';
import { render, screen } from 'test-utils';
import NavBar from './NavBar';

jest.mock('next-auth/react', () => ({
  useSession: jest.fn().mockReturnValue({
    data: {
      user: {
        email: 'test@test.com',
        image: 'testimage',
      },
    },
    status: 'not-loading',
  }),
  signOut: jest.fn(),
}));

const renderNavBar = (props) => {
  const utils = render(<NavBar {...props} />);
  const home = utils.container.firstChild;
  return { ...utils, home };
};

describe('NavBar', () => {
  it('should render and match snapshot', () => {
    const { home } = renderNavBar();
    expect(home).toBeTruthy();
    expect(home).toMatchSnapshot();
  });

  it('should render a signin button when no session data', () => {});
});
