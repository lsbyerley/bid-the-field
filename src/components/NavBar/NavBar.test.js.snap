// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavBar should render and match snapshot 1`] = `
<div
  class="navbar bg-base-100"
>
  <div
    class="navbar-start"
  >
    <div
      class="dropdown"
    >
      <label
        class="btn btn-sm btn-ghost lg:hidden"
        tabindex="0"
      >
        <svg
          aria-hidden="true"
          class="w-5 h-5"
          fill="none"
          stroke="currentColor"
          stroke-width="2"
          viewBox="0 0 24 24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M4 6h16M4 12h8m-8 6h16"
            stroke-linecap="round"
            stroke-linejoin="round"
          />
        </svg>
      </label>
      <ul
        class="p-2 mt-3 shadow menu menu-compact dropdown-content bg-base-100 rounded-box w-52"
        tabindex="0"
      >
        <li>
          <a
            class="active:bg-base-300"
            href="/"
          >
            Home
          </a>
        </li>
      </ul>
    </div>
    <a
      class="normal-case md:text-lg btn btn-sm btn-ghost"
      href="/"
    >
      Bid The Field
    </a>
  </div>
  <div
    class="hidden navbar-center lg:flex"
  />
  <div
    class="navbar-end"
  >
    <button
      aria-label="Toggle Dark Mode"
      class="btn btn-square btn-ghost btn-sm"
      role="switch"
      type="button"
    >
      <svg
        aria-hidden="true"
        class="inline-block w-4 h-4 stroke-current md:w-6 md:h-6"
        fill="none"
        stroke="currentColor"
        stroke-width="2"
        viewBox="0 0 24 24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M12 3v1m0 16v1m9-9h-1M4 12H3m15.364 6.364l-.707-.707M6.343 6.343l-.707-.707m12.728 0l-.707.707M6.343 17.657l-.707.707M16 12a4 4 0 11-8 0 4 4 0 018 0z"
          stroke-linecap="round"
          stroke-linejoin="round"
        />
      </svg>
    </button>
    <div
      class="ml-2 md:ml-4 dropdown dropdown-end"
    >
      <label
        class="btn btn-ghost btn-circle avatar"
        tabindex="0"
      >
        <div
          class="w-8 rounded-full md:w-10"
        >
          <img
            referrerpolicy="no-referrer"
            src="testimage"
          />
        </div>
      </label>
      <ul
        class="p-2 mt-3 shadow menu menu-compact dropdown-content bg-base-100 rounded-box w-52"
        tabindex="0"
      >
        <li
          class="menu-title"
        >
          <div>
            test@test.com
          </div>
        </li>
        <li>
          <a
            class=" active:bg-gray-500"
            href="/api/auth/signout"
          >
            Sign out
          </a>
        </li>
      </ul>
    </div>
  </div>
</div>
`;
